plugin/src/templates/edge/runtime.ts

```diff
@@ -38,13 +38,28 @@ const handler = async (req: Request, context: Context) => {
     return
   }
 
+  const geo = {
+    country: context.geo.country?.code,
+    region: context.geo.subdivision?.code,
+    city: context.geo.city,
+  }
+
+  // The geo object is passed through to the middleware unchanged
+  // so we're smuggling the Request and Netlify context object inside it
+
+  Object.defineProperty(geo, '__nf_context', {
+    value: context,
+    enumerable: false,
+  })
+
+  Object.defineProperty(geo, '__nf_request', {
+    value: req,
+    enumerable: false,
+  })
```

Comment:
Smuggling it in on `geo` for the proof of concept works, but if we ship it, could we not add '__nf_context' as a header that can be deserialized when consumed or somewhere else?

Reply:
It's certainly a hack and is undoubtedly fragile. Using a header would be ideal, but it's not serialisable as a string. Context is a class and we need access to the instance methods. `geo` is the only object that is passed-through from outside the EF. The alternative I've been looking at is adding it to a global `Map` (or `WeakMap`), and then passing the key in as a header.

Reply:
I've switched it to a global map, keyed by request ID. This is less likely to be broken by future changes in Next.js